apps-api/backoffice/src/modules/topic/repository.ts

```diff
@@ -0,0 +1,82 @@
+import Elysia from 'elysia'
+
+import { PrismaService, PrismaServicePlugin } from '../../plugins/prisma'
+import { fromPrismaPromise } from '../../utils/prisma'
+
+export class TopicRepository {
+  constructor(private readonly prismaService: PrismaService) {}
+  async getTopicById(topicId: string) {
+    return fromPrismaPromise(
+      this.prismaService.topic.findFirstOrThrow({
+        where: {
+          id: topicId,
+        },
+      })
+    )
+  }
+
+  async getTopics() {
+    return fromPrismaPromise(
+      this.prismaService.topic.findMany({
+        include: {
+          hashTagInTopics: {
+            include: {
+              hashTag: true,
+            },
+          },
+        },
+      })
+    )
+  }
+
+  async getUserFollowedTopics(userId: string) {
+    return fromPrismaPromise(
+      this.prismaService.userFollowsTopic.findMany({
+        where: {
+          userId: { equals: userId },
+        },
+        include: {
+          topic: {
+            include: {
+              hashTagInTopics: {
+                include: {
+                  hashTag: true,
+                },
+              },
+            },
+          },
+        },
+      })
+    )
+  }
+
+  async createUserFollowTopic(userId: string, topicId: string) {
+    return fromPrismaPromise(
+      this.prismaService.userFollowsTopic.create({
+        data: {
+          userId: userId,
+          topicId: topicId,
+        },
+      })
+    )
+  }
+
+  async deleteUserFollowTopic(userId: string, topicId: string) {
+    return fromPrismaPromise(
+      this.prismaService.userFollowsTopic.delete({
+        where: {
+          userId_topicId: {
+            userId: userId,
+            topicId: topicId,
+          },
+        },
+      })
+    )
+  }
+}
```

Comment:
I think we need to update `numberOfFollowingTopics` too and it should be done in one query

PS. This suggestion might have weird error when attempt to delete user follows topic, I will responsible for fixing it later

```suggestion
  async createUserFollowTopic(userId: string, topicId: string) {
    return fromPrismaPromise(
	  this.prismaService.user.update({
	    where: {
	      id: userId,
	    },
	    data: {
	      followedTopics: {
	        create: {
	          userId_topicId: {
	            userId: userId,
	            topicId: topicId,
	          },
	        },
	      },
	      numberOfFollowingTopics: {
	        increment: 1,
	      },
	    },
	  })
	)
  }

  async deleteUserFollowTopic(userId: string, topicId: string) {
    return fromPrismaPromise(
      this.prismaService.user.update({
        where: {
          id: userId,
        },
        data: {
          numberOfFollowingTopics: {
            decrement: 1,
          },
          followedTopics: {
            delete: {
              userId_topicId: {
                userId: userId,
                topicId: topicId,
              },
            },
          },
        },
      })
    )
  }
}
```